shared/js/textChatView.jsx

```diff
@@ -198,11 +198,26 @@ loop.shared.views.chat = (function(mozL10n) {
       var entriesClasses = classNames({
         "text-chat-entries": true
       });
+      var hideHelloWelcome = this.props.messageList.length > 0 ||
+        loop.shared.utils.isDesktop();
+
+      if (this.props.messageList.length > 0) {
+        var msgEntry = this.props.messageList[0];
```

Comment:
It probably is generally the first name, but maybe not 100% guaranteed. Having looked at this again, I think we should do a small refactor - I was never too happy with this code (I think I wrote it iirc!), but this is just getting a bit complex and weird now.

I think what we should do is:
- In `textChatStore`, stop storing the room name as part of the message list, instead store it as a separate item in the store.
- As a result, drop the `CHAT_CONTENT_TYPES.ROOM_NAME` type.
- In this file, we reduce down to having the `TextChatRoomName` always rendered at the top of the `text-chat-scroller`
- We pass in `this.state.roomName` (or whatever it gets called) as a prop from `TextChatView` into `TextChatEntriesView` and then into `TextChatRoomName`.
- `TextChatRoomName` then decides if to display the room name or fall back to the shortname.

I think that should simplify the code, especially with removing the special cases, and also help future maintenance.